airbyte-cdk/python/airbyte_cdk/sources/streams/http/http_error_handler.py

```diff
@@ -0,0 +1,129 @@
+import logging
+import requests
+from enum import Enum
+from typing import Any, Mapping, Optional
+from airbyte_cdk.utils.traced_exception import AirbyteTracedException
+from airbyte_cdk.models import FailureType
+
+class ResponseAction(Enum):
+
+    SUCCESS = "SUCCESS"
+    RETRY = "RETRY"
+    FAIL = "FAIL"
+    IGNORE = "IGNORE"
+
+class HttpErrorHandler():
+
+    error_mapping: Mapping[int, Mapping[str, Any]] = {
+        400: { "action": ResponseAction.FAIL, "failure_type": FailureType.config_error },
+        401: { "action": ResponseAction.FAIL, "failure_type": FailureType.config_error },
+        403: { "action": ResponseAction.FAIL, "failure_type": FailureType.config_error },
+        404: { "action": ResponseAction.FAIL, "failure_type": FailureType.system_error },
+        408: { "action": ResponseAction.RETRY, "failure_type": FailureType.transient_error },
+        429: { "action": ResponseAction.RETRY, "failure_type": FailureType.transient_error },
+        500: { "action": ResponseAction.RETRY, "failure_type": FailureType.transient_error },
+        502: { "action": ResponseAction.RETRY, "failure_type": FailureType.transient_error },
+        503: { "action": ResponseAction.RETRY, "failure_type": FailureType.transient_error },
```

Comment:
We should probably use the ResponseAction that fits [this logic](https://github.com/airbytehq/airbyte/blob/07db1caef2c4fcbf395abb8d418d61257e0c32ed/airbyte-cdk/python/airbyte_cdk/sources/streams/http/rate_limiting.py#L43-L45) to keep backward compatibility although I'm fine with retrying more (so 408 and 429 look like good default to me)

Reply:
Updated